### ISSUE TITLE
Kernel 5.4

### ISSUE THREAD

**Issue by prologic** [netdata/kernel-collector#8]

**Comment by prologic** [comment]
So its this kernel our build scripts fail with:

```sh
Successfully tagged kernel-collector:latest

cd ../kernel/ && make all;
make[1]: Entering directory '/kernel-collector/kernel'
clang -fno-stack-protector -S -nostdinc -I/usr/src/linux/arch/x86/include/generated -I/usr/src/linux/arch/x86/include/generated/uapi -I/usr/src/linux/include -I/usr/src/linux/include/generated/uapi -I/usr/src/linux/arch/x86/include -I/usr/src/linux/arch/x86/include/uapi -I/usr/src/linux/arch/x86/include/generated -I/usr/src/linux/arch/x86/include/generated/uapi -I/usr/src/linux/include -I/usr/src/linux/include/uapi -include /usr/src/linux/include/linux/kconfig.h -I../includes -I/usr/lib/clang/9.0.0/include/ \
	-D__KERNEL__ -D__ASM_SYSREG_H -Wno-unused-value -Wno-pointer-sign \
    	-Wno-compare-distinct-pointer-types \
    	-Wno-gnu-variable-sized-type-not-at-end \
    	-Wno-tautological-compare \
	-DNETDATASEL=0 \
	-O2 -emit-llvm -c process_kern.c
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:12:
In file included from /usr/src/linux/arch/x86/include/asm/page.h:12:
/usr/src/linux/arch/x86/include/asm/page_64.h:49:2: error: expected '(' after 'asm'
        alternative_call_2(clear_page_orig,
        ^
/usr/src/linux/arch/x86/include/asm/alternative.h:256:2: note: expanded from macro 'alternative_call_2'
        asm_inline volatile (ALTERNATIVE_2("call %P[old]", "call %P[new1]", feature1,\
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
In file included from /usr/src/linux/arch/x86/include/asm/processor.h:12:
In file included from /usr/src/linux/arch/x86/include/asm/math_emu.h:5:
In file included from /usr/src/linux/arch/x86/include/asm/ptrace.h:5:
/usr/src/linux/arch/x86/include/asm/segment.h:266:2: error: expected '(' after 'asm'
        alternative_io ("lsl %[seg],%[p]",
        ^
/usr/src/linux/arch/x86/include/asm/alternative.h:240:2: note: expanded from macro 'alternative_io'
        asm_inline volatile (ALTERNATIVE(oldinstr, newinstr, feature)   \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
In file included from /usr/src/linux/arch/x86/include/asm/processor.h:24:
/usr/src/linux/arch/x86/include/asm/special_insns.h:205:2: error: expected '(' after 'asm'
        alternative_io(".byte " __stringify(NOP_DS_PREFIX) "; clflush %P0",
        ^
/usr/src/linux/arch/x86/include/asm/alternative.h:240:2: note: expanded from macro 'alternative_io'
        asm_inline volatile (ALTERNATIVE(oldinstr, newinstr, feature)   \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:540:17: warning: taking address of packed member 'sp0' of class or structure 'x86_hw_tss' may result in an
      unaligned pointer value [-Waddress-of-packed-member]
        this_cpu_write(cpu_tss_rw.x86_tss.sp0, sp0);
                       ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/include/linux/percpu-defs.h:509:68: note: expanded from macro 'this_cpu_write'
#define this_cpu_write(pcp, val)        __pcpu_size_call(this_cpu_write_, pcp, val)
                                                                          ^~~
/usr/src/linux/include/linux/percpu-defs.h:376:22: note: expanded from macro '__pcpu_size_call'
        __verify_pcpu_ptr(&(variable));                                 \
                            ^~~~~~~~
/usr/src/linux/include/linux/percpu-defs.h:220:47: note: expanded from macro '__verify_pcpu_ptr'
        const void __percpu *__vpp_verify = (typeof((ptr) + 0))NULL;    \
                                                     ^~~
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:557:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an
      unaligned pointer value [-Waddress-of-packed-member]
        return this_cpu_read_stable(cpu_current_top_of_stack);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/processor.h:384:34: note: expanded from macro 'cpu_current_top_of_stack'
#define cpu_current_top_of_stack cpu_tss_rw.x86_tss.sp1
                                 ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/percpu.h:392:59: note: expanded from macro 'this_cpu_read_stable'
#define this_cpu_read_stable(var)       percpu_stable_op("mov", var)
                                                                ^~~
/usr/src/linux/arch/x86/include/asm/percpu.h:219:16: note: expanded from macro 'percpu_stable_op'
                    : "p" (&(var)));                    \
                             ^~~
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:557:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an
      unaligned pointer value [-Waddress-of-packed-member]
        return this_cpu_read_stable(cpu_current_top_of_stack);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/processor.h:384:34: note: expanded from macro 'cpu_current_top_of_stack'
#define cpu_current_top_of_stack cpu_tss_rw.x86_tss.sp1
                                 ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/percpu.h:392:59: note: expanded from macro 'this_cpu_read_stable'
#define this_cpu_read_stable(var)       percpu_stable_op("mov", var)
                                                                ^~~
/usr/src/linux/arch/x86/include/asm/percpu.h:224:16: note: expanded from macro 'percpu_stable_op'
                    : "p" (&(var)));                    \
                             ^~~
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:557:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an
      unaligned pointer value [-Waddress-of-packed-member]
        return this_cpu_read_stable(cpu_current_top_of_stack);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/processor.h:384:34: note: expanded from macro 'cpu_current_top_of_stack'
#define cpu_current_top_of_stack cpu_tss_rw.x86_tss.sp1
                                 ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/percpu.h:392:59: note: expanded from macro 'this_cpu_read_stable'
#define this_cpu_read_stable(var)       percpu_stable_op("mov", var)
                                                                ^~~
/usr/src/linux/arch/x86/include/asm/percpu.h:229:16: note: expanded from macro 'percpu_stable_op'
                    : "p" (&(var)));                    \
                             ^~~
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:557:30: warning: taking address of packed member 'sp1' of class or structure 'x86_hw_tss' may result in an
      unaligned pointer value [-Waddress-of-packed-member]
        return this_cpu_read_stable(cpu_current_top_of_stack);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/processor.h:384:34: note: expanded from macro 'cpu_current_top_of_stack'
#define cpu_current_top_of_stack cpu_tss_rw.x86_tss.sp1
                                 ^~~~~~~~~~~~~~~~~~~~~~
/usr/src/linux/arch/x86/include/asm/percpu.h:392:59: note: expanded from macro 'this_cpu_read_stable'
#define this_cpu_read_stable(var)       percpu_stable_op("mov", var)
                                                                ^~~
/usr/src/linux/arch/x86/include/asm/percpu.h:234:16: note: expanded from macro 'percpu_stable_op'
                    : "p" (&(var)));                    \
                             ^~~
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:795:2: error: expected '(' after 'asm'
        alternative_input(BASE_PREFETCH, "prefetchnta %P1",
        ^
/usr/src/linux/arch/x86/include/asm/alternative.h:221:2: note: expanded from macro 'alternative_input'
        asm_inline volatile (ALTERNATIVE(oldinstr, newinstr, feature)   \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
In file included from /usr/src/linux/include/linux/thread_info.h:38:
In file included from /usr/src/linux/arch/x86/include/asm/thread_info.h:53:
In file included from /usr/src/linux/arch/x86/include/asm/cpufeature.h:5:
/usr/src/linux/arch/x86/include/asm/processor.h:807:2: error: expected '(' after 'asm'
        alternative_input(BASE_PREFETCH, "prefetchw %P1",
        ^
/usr/src/linux/arch/x86/include/asm/alternative.h:221:2: note: expanded from macro 'alternative_input'
        asm_inline volatile (ALTERNATIVE(oldinstr, newinstr, feature)   \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
/usr/src/linux/include/linux/thread_info.h:134:2: error: expected '(' after 'asm'
        WARN(1, "Buffer overflow detected (%d < %lu)!\n", size, count);
        ^
/usr/src/linux/include/asm-generic/bug.h:124:3: note: expanded from macro 'WARN'
                __WARN_printf(TAINT_WARN, format);                      \
                ^
/usr/src/linux/include/asm-generic/bug.h:93:3: note: expanded from macro '__WARN_printf'
                __WARN_FLAGS(BUGFLAG_NO_CUT_HERE | BUGFLAG_TAINT(taint));\
                ^
/usr/src/linux/arch/x86/include/asm/bug.h:79:2: note: expanded from macro '__WARN_FLAGS'
        _BUG_FLAGS(ASM_UD2, BUGFLAG_WARNING|(flags));           \
        ^
/usr/src/linux/arch/x86/include/asm/bug.h:35:2: note: expanded from macro '_BUG_FLAGS'
        asm_inline volatile("1:\t" ins "\n"                             \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
In file included from /usr/src/linux/include/linux/spinlock.h:51:
In file included from /usr/src/linux/include/linux/preempt.h:78:
In file included from /usr/src/linux/arch/x86/include/asm/preempt.h:7:
/usr/src/linux/include/linux/thread_info.h:150:6: error: expected '(' after 'asm'
        if (WARN_ON_ONCE(bytes > INT_MAX))
            ^
/usr/src/linux/include/asm-generic/bug.h:98:3: note: expanded from macro 'WARN_ON_ONCE'
                __WARN_FLAGS(BUGFLAG_ONCE |                     \
                ^
/usr/src/linux/arch/x86/include/asm/bug.h:79:2: note: expanded from macro '__WARN_FLAGS'
        _BUG_FLAGS(ASM_UD2, BUGFLAG_WARNING|(flags));           \
        ^
/usr/src/linux/arch/x86/include/asm/bug.h:35:2: note: expanded from macro '_BUG_FLAGS'
        asm_inline volatile("1:\t" ins "\n"                             \
        ^
/usr/src/linux/include/linux/compiler_types.h:210:24: note: expanded from macro 'asm_inline'
#define asm_inline asm __inline
                       ^
In file included from process_kern.c:2:
In file included from /usr/src/linux/include/linux/bpf.h:9:
In file included from /usr/src/linux/include/linux/workqueue.h:9:
In file included from /usr/src/linux/include/linux/timer.h:6:
In file included from /usr/src/linux/include/linux/ktime.h:24:
In file included from /usr/src/linux/include/linux/time.h:6:
In file included from /usr/src/linux/include/linux/seqlock.h:36:
/usr/src/linux/include/linux/spinlock.h:60:10: fatal error: 'asm/mmiowb.h' file not found
#include <asm/mmiowb.h>
         ^~~~~~~~~~~~~~
5 warnings and 8 errors generated.
make[1]: *** [Makefile:36: process_kern.o] Error 1
make[1]: Leaving directory '/kernel-collector/kernel'
make: *** [Makefile:28: ../kernel/process_kern.o] Error 2
ERROR: Build failed ...
Dropping into a shell ...
/kernel-collector #
/kernel-collector #
```

**Comment by thiagoftsm** [comment]
About the `inline`, please, try to comment inside ```/lib/modules/5.4.17/build/include/generated/autoconf.h```  the line ```#define CONFIG_CC_HAS_ASM_INLINE 1```.
For the not found, it lives inside ```/usr/src/linux-5.4.17/arch/x86/include/generated/```.

**Comment by prologic** [comment]
@thiagoftsm Can you suggest a patch to the current master that _might_ fix this? I've tried a few things and don't work :/

**Comment by prologic** [comment]
Here's how to repro:

```sh
$ docker build --build-arg KERNEL_VERSION=5.4.18 -t kernel-collector .
docker run -i-t --rm -v $PWD:/kernel-collector kernel-collector
```

**Comment by thiagoftsm** [comment]
> @thiagoftsm Can you suggest a patch to the current master that _might_ fix this? I've tried a few things and don't work :/

I will try to replicate to help you!

**Comment by prologic** [comment]
Fixed by #36
